channel/channel.go

```diff
@@ -321,3 +324,22 @@ func (c *Channel) SignAndAddState(s state.State, sk *[]byte) (state.SignedState,
 	}
 	return ss, nil
 }
+
+// UpdateWithChainEvent mutates the receiver if provided with a "new" chain event (with a greater block number than previously seen)
+func (c *Channel) UpdateWithChainEvent(event chainservice.Event) (*Channel, error) {
+	if event.BlockNum() <= c.latestBlockNumber {
+		return c, nil // ignore stale information TODO: is this reorg safe?
```

Comment:
Without too much thought: a `<` instead of `<=` makes it reorg safe.

As is, two events fired in the same block won't both get processed.